src/components/c-help-redirect.ts

```diff
@@ -0,0 +1,258 @@
+        assert(command.channel);
+        assert(command.channel instanceof Discord.GuildChannel);
+
+        // Only allowed in #c-help-text
+        if (command.channel.id != this.wheatley.channels.c_help_text.id) {
+            await command.reply(`Can only be used in <#${this.wheatley.channels.c_help_text.id}>`, true);
+            return;
+        }
+
+        //for manual triggers, trust the caller and don't check the message
+        //supposedly the automatic check didn't trigger, so checking the message again would fail again
+        if (user) {
+            await command.channel.send(
+                `<@${user.id}> Your code looks like C++ code, but this is a C channel. ` +
+                    `Did you mean to post in <#${this.wheatley.channels.cpp_help_text.id}>?`,
+            );
+        } else {
+            await command.channel.send(
+                `This code looks like C++ code, but this is a C channel. ` +
+                    `Did you mean to post in <#${this.wheatley.channels.cpp_help_text.id}>?`,
+            );
+        }
+    }
+
+    async not_cpp(command: TextBasedCommand, user: Discord.User | null) {
+        assert(command.channel);
+        assert(command.channel instanceof Discord.GuildChannel);
+
+        // Only allowed in #cpp-help-text
+        if (command.channel.id != this.wheatley.channels.cpp_help_text.id) {
+            await command.reply(`Can only be used in <#${this.wheatley.channels.cpp_help_text.id}>`, true);
+            return;
+        }
+
+        //for manual triggers, trust the caller and don't check the message
+        //supposedly the automatic check didn't trigger, so checking the message again would fail again
+        if (user) {
+            await command.channel.send(
+                `<@${user.id}> Your code looks like C code, but this is a C++ channel. ` +
+                    `Did you mean to post in <#${this.wheatley.channels.c_help_text.id}>?`,
+            );
+        } else {
+            await command.channel.send(
+                `This code looks like C code, but this is a C++ channel. ` +
+                    `Did you mean to post in <#${this.wheatley.channels.c_help_text.id}>?`,
+            );
+        }
+    }
+
+    override async on_message_create(message: Discord.Message) {
+        // Ignore self, bots, and messages outside TCCPP (e.g. dm's)
+        if (
+            message.author.id == this.wheatley.client.user!.id ||
+            message.author.bot ||
+            message.guildId != this.wheatley.TCCPP.id
+        ) {
+            return;
+        }
+
+        //only auto-check new members
+        if (await this.is_not_new_member(message)) {
+            return;
+        }
+
+        // Only check messages in help-text channels
+        if (message.channel.id == this.wheatley.channels.c_help_text.id) {
+            if (this.check_message_for_cpp_code(message)) {
+                await message.reply(
```

Comment:
It might be good to set a timeout to not spam users, e.g. only send this once an hour. The `SelfClearingSet` utility is a good way to go, happy to help with implementing this.

Reply:
I think I figured it out its usage on other places.

Reply:
Sweet!